nimble/host/include/host/ble_audio_broadcast.h

```diff
@@ -0,0 +1,271 @@
+ * changed during it's lifetime. To modify BASE configuration destroy it using
+ * `ble_audio_broadcast_destroy` and recreate with `ble_audio_broadcast_create`
+ *
+ * @param[in] params            Pointer to structure with new advertising data
+ * @return                      0 on success;
+ *                              A non-zero value on failure.
+ */
+int ble_audio_broadcast_update(const struct ble_broadcast_update_params
+                               *params);
+
+/** BIG Subgroup parameters */
+struct ble_broadcast_subgroup_params {
+    /** Subgroup level Codec information */
+    struct ble_audio_codec_id *codec_info;
+
+    /** Subgroup level Codec Specific Configuration */
+    uint8_t *codec_spec_config;
+
+    /** Subgroup level Codec Specific Configuration length */
+    uint8_t codec_spec_config_len;
+
+    /** Subgroup Metadata */
+    uint8_t *metadata;
+
+    /** Subgroup Metadata length*/
+    uint8_t metadata_len;
+};
+
+/**
+ * @brief Build BIG subgroup structure
+ *
+ * This is a helper function can be used to fill out `ble_broadcast_subgroup`
+ * structure. Created subgroup extends subgroup list in provided BASE.
+ * This function increases `num_subgroups` in BASE structure.
+ *
+ * @param[out] base             Pointer to a `ble_audio_base` structure,
+ *                              that will be extended by the new subgroup
+ * @param[out] sub              Pointer to a `ble_broadcast_subgroup`
+ *                              structure, that will be filled out with
+ *                              supplied configuration
+ * @param[in] params            Pointer to a `ble_broadcast_subgroup_params`
+ *                              structure, containing information about new
+ *                              subgroup
+ *
+ * @return                      0 on success;
+ *                              A non-zero value on failure.
+ */
+int ble_audio_broadcast_build_sub(struct ble_audio_base *base,
+                                  struct ble_audio_subgroup *sub,
+                                  const struct ble_broadcast_subgroup_params
+                                  *params);
+
+/** BIS parameters */
+struct ble_broadcast_bis_params {
+    /** BIS index */
+    uint8_t idx;
+
+    /** BIS level Codec Specific Configuration */
+    uint8_t *codec_spec_config;
+
+    /** BIS level Codec Specific Configuration length */
+    uint8_t codec_spec_config_len;
+};
+
+/**
+ * @brief Build BIS structure
+ *
+ * This is a helper function can be used to fill out `ble_broadcast_bis`
+ * structure. Created BIS extends BIS list in provided subgroup.
+ * This function increases `bis_cnt` in subgroup structure.
+ *
+ * @param[out] sub              Pointer to a `ble_broadcast_subgroup`
+ *                              structure,
+ *                              that will be extended by the new BIS
+ * @param[out] bis              Pointer to a `ble_broadcast_bis`
```

Comment:
I'd add line "In case of error, filled out data may be erroneous.". Pointers do not change value, so they are ok - but data pointed by them will contain invalid values. Would that be ok?

Reply:
It depends on what is the common way to handle it across nimble. Is this a common solution ?

Reply:
The way that's implemented right now, it has no true way to "fail", because all that happens there is memcpy and STAILQ_INSERT_HEAD/TAIL, plus some fields are set. So we always return 0. Description can be made future proof, as it is now, so in the future some error checking might take place. Maybe this comment should be ommited for now, as this function weakest point is memcpy, and we cannot really check this for error. Any thoughts @rymanluk @sjanc ?

Reply:
Imho, this is exactly why the `return` is the result of operation. In a case of error returned, we don't care about [out] params and its state is unknown. Current description should be  good.